test/e2e_node/util.go

```diff
@@ -138,6 +138,12 @@ func setKubeletConfiguration(f *framework.Framework, kubeCfg *kubeletconfig.Kube
 	const (
 		restartGap   = 40 * time.Second
 		pollInterval = 5 * time.Second
+
+		// as we use period=10*time.Second and jitterFactor=0.2
+		// in cc.syncConfigSource(client, nodeName),
+		// so we set waitGap = 12*time.Second to wait for kubelet restart
+		// ref: pkg/kubelet/kubeletconfig/controller.go#L220-L222
+		waitGap = 12 * time.Second
```

Comment:
We already wait for up to 40 seconds, which is plenty of time in practice. I don't see the need to tack on a sleep.
```
// poll for new config, for a maximum wait of restartGap
Eventually(func() error {
	...
}, restartGap, pollInterval).Should(BeNil())
```

Reply:
@mtaufen Hi, thanks!
After reading the code, i think `restartGap` is the timeout for the assertion in `Eventually()`. If assertion passes, it will return. See more comments [here](https://github.com/kubernetes/kubernetes/issues/57701#issuecomment-354707445).